lib/rubocop/cop/capybara/mixin/css_attributes_parser.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Capybara
+      # Css selector parser.
+      # @api private
+      class CssAttributesParser
+        def initialize(selector)
+          @selector = selector
+          @state = :initial
+          @temp = ''
+          @results = {}
+          @bracket_count = 0
+        end
+
+        # @return [Array<String>]
+        def parse # rubocop:disable Metrics/MethodLength
+          @selector.chars do |char|
+            if char == '['
+              on_bracket_start(char)
+              next
+            elsif char == ']'
+              on_bracket_end(char)
+              next
+            end
+            @temp += char if @state == :inside_attr
```

Comment:
You can remove the `next`s:
```suggestion
            elsif char == ']'
              on_bracket_end(char)
            else
              @temp += char if @state == :inside_attr
            end
```